lib/rb-inotify/notifier.rb

```diff
@@ -292,7 +294,11 @@ def read_events
     # Same as IO#readpartial, or as close as we need.
     def readpartial(size)
       # Use Ruby's readpartial if possible, to avoid blocking other threads.
-      return to_io.readpartial(size) if self.class.supports_ruby_io?
+      begin
+        return to_io.readpartial(size) if self.class.supports_ruby_io?
+      rescue Errno::EBADF
```

Comment:
Add a comment explaining when this can happen and why returning `nil` is correct.